cli/src/commands/git/push.rs

```diff
@@ -393,6 +393,27 @@ pub fn cmd_git_push(
             ),
             remote = remote.as_symbol()
         );
+
+        if bookmark_updates.is_empty() {
```

Comment:
Can you include in the commit message why you want this hint?

If your intuition was that `jj git push` (with no arguments) would push deleted bookmarks, it doesn't make sense to mention deleted bookmarks only if there are no other updates.

Reply:
What happened was that I created a local bookmark that pointed to `@-`, but then wanted to delete it. So I did that, then `jj git push`ed, assuming it would push since it was one of `@`'s ancestors, but then forgot that I needed to pass `--delete`.

I had that case in mind, which is why I did that. You're right that it would probably be better to do it unconditionally.

Reply:
How's that for the commit message, does it explain why I think this is useful? I can add more detail, it's not the easiest to articulate :)

Reply:
> What happened was that I created a local bookmark that pointed to `@-`, but then wanted to delete it. So I did that, then `jj git push`ed,

In that case, maybe we can look for remote bookmarks to be deleted within the default push revset range (plus the remote bookmarks themselves.) It won't help if the deleted local bookmarks had been rewritten/rebased, but can greatly reduce the amout of false positives.

Reply:
> In that case, maybe we can look for remote bookmarks to be deleted within the default push revset range (plus the remote bookmarks themselves.)

Won't we have to look at a previous operation for that, since the default push revset doesn't match in this case?

Reply:
> greatly reduce the amout of false positives

How would this lead to false positives? I'm unaware.

Reply:
> Won't we have to look at a previous operation for that,

I meant something like `ancestors(remote_bookmarks(..)..@, 2) & <remote bookmarks to be deleted>`.

> How would this lead to false positives? I'm unaware.

There's no clue to assume that the user intended to push deleted bookmarks, right? It seems just wrong to list random deleted bookmarks.

Reply:
> I meant something like `ancestors(remote_bookmarks(..)..@, 2) & <remote bookmarks to be deleted>`.

Hmm... yeah, maybe that might work well. Perhaps the message should be something like `Hint: You can delete X nearby remote bookmark(s) (and Y additional ones) with jj git push --deleted` or something.

> There's no clue to assume that the user intended to push deleted bookmarks, right? It seems just wrong to list random deleted bookmarks.

Fair. Well, they are tracked, so I'd imagine there's a good chance they want to? (And if they don't, then they'll be deleted as soon as they pass `--deleted`.

Reply:
Though I am now questioning the usefulness of this 😅 I'll defer to you on that end.

Reply:
Since the default isn't `jj git push --all`, I don't think it's great idea to list "all" deleted bookmarks.

> Though I am now questioning the usefulness of this 😅 I'll defer to you on that end.

I personally don't think this hint will help. It's not easy to guess which deleted bookmarks would belong to the current push target (if they weren't deleted), and it's kidna obvious that `jj git push` doesn't push deleted bookmarks unless specified.